### PR TITLE
Add tests for Quarkus live reload

### PR DESCRIPTION
**Pull request by jsmrcka**
Covers:
https://issues.redhat.com/browse/QUARKUS-1008
https://issues.redhat.com/browse/QUARKUS-1020

### CODE REVIEW

File: testsuite/src/it/java/io/quarkus/ts/startstop/LiveReloadTest.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+import io.quarkus.ts.startstop.utils.Apps;
+import io.quarkus.ts.startstop.utils.HttpUtils;
+import io.quarkus.ts.startstop.utils.MvnCmds;
+
+@Tag("instrumentation")
```
Comment (jsmrcka) [inline testsuite/src/it/java/io/quarkus/ts/startstop/LiveReloadTest.java:36]: Do we want a new tag, which would have to be added to all relevant jenkins jobs?
Or use existing tag?

Comment (kshpak) [reply testsuite/src/it/java/io/quarkus/ts/startstop/LiveReloadTest.java:36]: IMO we don't need it

**Comment by jsmrcka** [comment]
Open questions:
- Is `StartStopTS` even the right place for this? 
- Create a new sigle-purpose application `app-live-reload` vs. use one of the existing ones with a couple of modifications (new endpoints).
- What tag to use to annotate the `LiveReloadTest`?

@rsvoboda Any thoughts?

**Review by rsvoboda — CHANGES_REQUESTED** [review]
Disable live reload instrumentation by default - https://docs.google.com/spreadsheets/d/1KCiamQ3YtxE8AVlr9lG8crWqGaKFVNjSZ0CAhc3jR3s/edit#gid=0 says `Yes (src zip test in marete)`

Added class files should trigger a full reload - https://github.com/quarkus-qe/quarkus-startstop/blob/main/testsuite/src/it/java/io/quarkus/ts/startstop/ArtifactGeneratorTest.java#L234 does modification of existing file, extend the code to also add new file

This `app-live-reload` effort is not necessary.

**Comment by jsmrcka** [comment]
Dropped in favor of https://github.com/quarkus-qe/quarkus-startstop/pull/85.

**Pull request closed without merging**